akka-stream/src/main/scala/akka/stream/RestartSettings.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream
+
+import scala.concurrent.duration.FiniteDuration
+
+import akka.util.JavaDurationConverters._
+
+final class RestartSettings private (
+    val minBackoff: FiniteDuration,
+    val maxBackoff: FiniteDuration,
+    val randomFactor: Double,
+    val maxRestarts: Int,
+    val maxRestartsWithin: FiniteDuration) {
+
+  /** Scala API: minimum (initial) duration until the child actor will started again, if it is terminated */
+  def withMinBackoff(value: FiniteDuration): RestartSettings = copy(minBackoff = value)
+
+  /** Java API: minimum (initial) duration until the child actor will started again, if it is terminated */
+  def withMinBackoff(value: java.time.Duration): RestartSettings = copy(minBackoff = value.asScala)
+
+  /** Scala API: the exponential back-off is capped to this duration */
+  def withMaxBackoff(value: FiniteDuration): RestartSettings = copy(maxBackoff = value)
+
+  /** Java API: the exponential back-off is capped to this duration */
+  def withMaxBackoff(value: java.time.Duration): RestartSettings = copy(maxBackoff = value.asScala)
+
+  /**
+   * After calculation of the exponential back-off an additional random delay based on this factor is added
+   * e.g. `0.2` adds up to `20%` delay.  In order to skip this additional delay pass in `0`
+   */
+  def withRandomFactor(value: Double): RestartSettings = copy(randomFactor = value)
+
+  /** The amount of restarts is capped to this amount within a timeframe of [[maxRestartsWithin]] */
+  def withMaxRestarts(value: Int): RestartSettings = copy(maxRestarts = value)
+
+  /** Scala API: the duration after which to reset the restart count and current exponential backoff */
+  def withMaxRestartsWithin(value: FiniteDuration): RestartSettings = copy(maxRestartsWithin = value)
```

Comment:
since maxRestarts and maxRestartsWithin are related and should be set together I'd suggest to only have

`def withMaxRestarts(count: Int, within: FiniteDuration)`

Reply:
After updating some of the docs, got a small question on this. Because this `maxRestartsWithin` is being used to determine when the backoff time resets back to `minBackoff` it isn't possible to configure this reset time without also setting `maxRestarts`? Is there a use case for wanting a separate `backoffResetAfter: FiniteDuration` which can be configured without needing to limit `maxRestarts`? That value could then be reused as what the `maxRestartsWithin` is doing? Or is this all a separate issue.